src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidAgeException;
+import core.basesyntax.exception.LoginValidateException;
+import core.basesyntax.exception.PasswordValidateException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
+    private static final int MINIMAL_LOGIN_LENGTH = 4;
+    private static final int FIRST_LOGIN_CHARACTER_INDEX = 0;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkExistUserInDataBase(user.getLogin());
+        ageValidation(user.getAge());
+        validatePassword(user.getPassword());
+        return storageDao.add(user);
+    }
+
+    private void loginValidation(String login) {
+        if (login == null || login.length() <= MINIMAL_LOGIN_LENGTH) {
+            throw new LoginValidateException("your login must be more than 4 characters");
+        } else if (!Character.isLetter(login.charAt(FIRST_LOGIN_CHARACTER_INDEX))) {
+            throw new LoginValidateException("your login should start from the letter");
```

Comment:
this was not part of the task, as a software engineers we should be mindfull of of solving and implementing business problems, but our own additions should be talked through first, before addition